contracts/claim/src/contract.rs

```diff
@@ -19,9 +22,20 @@ pub fn instantiate(
     deps: DepsMut,
     _env: Env,
     _info: MessageInfo,
-    _msg: InstantiateMsg,
+    msg: InstantiateMsg,
 ) -> Result<Response, ContractError> {
+    let marketplace =
+        MarketplaceContract(deps.api.addr_validate(&msg.marketplace_addr).map_err(|_| {
```

Comment:
why does this need to depend on sg_marketplace and not just be an address ? imo claim contract shouldn't care as long as it has an "admin"

Reply:
Adding to this, would be nice to have a generalized contract addr checker

Reply:
@jhernandezb It doesn't need to be but I wanted to start using contract helpers within contracts instead of raw messages for convenience. This contract depends on sg_marketplace for the hook anyway. It is a bit redundant to use the helper here so can remove it if you feel strongly about it.

Reply:
sounds good to me